compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -424,13 +424,30 @@ class DependencyRecorder {
       classDependencies.foreach(recordClassDependency(cb, _))
     clear()
 
+  private val _siblingClassfiles = new mutable.HashMap[PlainFile, Path]
+
+  /**Constructs a sibling class to the `jpath`.
+   * Does not validate if it exists as a real file.
+   * The way this works is that by the end of compilation analysis,
+   * there should be a corresponding NonLocalClass sent to zinc with the same class file name.
```

Comment:
Mention `generatedNonLocalClass` by name (also this is only needed for non-library dependencies)

Reply:
done